Objects/dictobject.c

```diff
@@ -5491,10 +5491,9 @@ dictiter_iternext_threadsafe(PyDictObject *d, PyObject *self,
     Py_DECREF(d);
     return -1;
 
-    int res;
 try_locked:
     Py_BEGIN_CRITICAL_SECTION(d);
-    res = dictiter_iternextitem_lock_held(d, self, out_key, out_value);
+    int res = dictiter_iternextitem_lock_held(d, self, out_key, out_value);
```

Comment:
`Py_BEGIN_CRITICAL_SECTION` declares a scope, so `int res` needs to be declared outside of it. However, you can't write:

```
try_locked:
    int res;
    ...
```

in C (until C23, I think) because labels have to be on statements (not declarations). But you can instead write:

```
try_locked: ;
    int res;
    ...
```

I don't see the original warning on macOS with Clang version 15.0.0 nor do I understand it. What compile options are you using?

Reply:
`./configure --disable-gil --enable-experimental-jit`